### DISCUSSION TITLE
LLM functionality for trilium

### DISCUSSION THREAD

**Discussion by harishpillay** [TriliumNext/Trilium#4506] [Ideas]
I am wondering is anyone is looking at LLM functionality for trilium? I just came across a plugin for Obsidian (https://github.com/hinterdupfinger/obsidian-ollama) and thought it would be interesting to have that sort of functionality become available for trilium as well. 

PS:  I just moved to trilium from joplin and I feel this is the right note-taking tool for me.

**Comment by zerebos** [comment]
There's an existing plugin to integrate with ChatGPT here: https://github.com/soulsands/trilium-chat

But doing something like that plugin for Trilium is definitely possible but I don't think one that acts exactly like that exists.

**Reply by harishpillay** [reply]
Thanks for that link. I will explore it. 

I am inclined not to have to go out to the an external entity but do something like how llamafile does - https://github.com/Mozilla-Ocho/llamafile/.

Thanks again.

**Comment by zadam** [comment]
Integrating LLM support into Trilium would be great, but currently it means tying it to some specific provider / API and that seems quite unwise.

**Comment by sottey** [comment]
I agree with this. There are some shim patterns out there that are being developed, but configuring them in Trilium could be problematic or at least a lot of work.

It would be super nice, but I think the best solution would be an external software that can get readonly access to the Trilium SQLite db as a data source.

My .02

**Comment by harishpillay** [comment]
I was looking at llamafile (https://hacks.mozilla.org/2023/11/introducing-llamafile/) that is a model you have locally and then have it finetuned against the Trilium db.

**Comment by tan-yong-sheng** [comment]
perhaps litellm + ollama?

> LiteLLM provides a unified interface to call 100+ LLMs using the same Input/Output format, including OpenAI, Huggingface, Anthropic, vLLM, Cohere, etc: https://github.com/BerriAI/litellm/tree/main?tab=readme-ov-file#-proxy-endpoints---swagger-docs

**Comment by blargg** [comment]
LLM features were added. See the discussion here: https://github.com/TriliumNext/Trilium/issues/5543
It should be in version v0.94.0 and up.
